### ISSUE TITLE
Export fileType type and tag constants

### ISSUE THREAD

**Issue by erikgeiser** [google/rpmpack#47]
In my opinion, the `fileType` type from `file_types.go` should be exported, so libraries can for example write wrappers around `RPM.AddFiles()` like this:
```go
func addFileWrapper(rpm *rpmpack.RPM, ..., fileType rpmpack.FileType) error {
    ...
    rpm.AddFile(rpmpack.RPMFile{
        ...
        Type: fileType
    })
}
```

Similarly, the tag constants from `tags.go` should also be exported to allow libraries to use them when extracting and validating RPM properties in tests via these tags.

[nfpm](https://github.com/goreleaser/nfpm/), which depends on `rpmpack`, currently has to employ an ugly workaround for `fileType` in such a wrapper and redefines many tag constants for tests.

**Comment by jarondl** [comment]
As for the tags, I am sorry to hear they are redefined many times. I did initially think people use them freely. But now we even have a method to add other tags:
https://pkg.go.dev/github.com/google/rpmpack?tab=doc#RPM.AddCustomTag
So yeah I am not opposed to exporting them.

I wonder how it would make the doc look though? Once we export them, do we need to say what they mean? Or is that obvious?

**Comment by erikgeiser** [comment]
The `RPM.AddCustomTag` method is great. I looked into it a bit more and I have changed my mind about the tags. As `rpmpack` already provides methods to populate all the tags it defines, the tags are only really useful to extract the tag data later on. However, this is not a usecase of `rpmpack`. In fact, `nfpm` uses github.com/sassoftware/go-rpmutils to do that. So I looked into `rpmutils` again and I found the tags to be exported there. So now I think you were right not to export them.

as the `FileType` PR is already merged, this issue can be closed in my opinion.
